packages/client/src/game/canvas/index.tsx

```diff
@@ -0,0 +1,284 @@
+	};
+
+	const splitImage = (
+		imgSrc: string,
+		parts: number,
+	): Promise<{
+		horizontal: HTMLImageElement;
+		vertical: HTMLImageElement;
+	}[]> => {
+		return new Promise((resolve) => {
+			const image = new Image();
+			image.src = imgSrc;
+			image.onload = () => {
+				const {width, height} = image;
+
+				const partWidth = Math.floor(width / parts);
+				const sprites = [] as {
+					horizontal: HTMLImageElement;
+					vertical: HTMLImageElement;
+				}[];
+
+				for (let i = 0; i < parts; i++) {
+					const startX = i * partWidth;
+
+					const toSprite = (vertical: boolean = false) => {
+						const canvas = document.createElement('canvas');
+						const canvasCtx = canvas.getContext('2d') as CanvasRenderingContext2D;
+						canvas.width = partWidth;
+						canvas.height = height;
+
+						if (vertical) {
+							canvasCtx.translate(partWidth / 2, height / 2);
+							canvasCtx.rotate((90 * Math.PI) / 180);
+							canvasCtx.drawImage(
+								image,
+								startX,
+								0,
+								partWidth,
+								height,
+								-partWidth / 2,
+								-height / 2,
+								partWidth,
+								height,
+							);
+						} else {
+							canvasCtx.drawImage(
+								image,
+								startX,
+								0,
+								partWidth,
+								height,
+								0,
+								0,
+								partWidth,
+								height,
+							);
+						}
+
+						const spriteImage = new Image();
+						spriteImage.src = canvas.toDataURL();
+						return spriteImage;
+					};
+					sprites.push({
+						horizontal: toSprite(false),
+						vertical: toSprite(true),
+					});
+				}
+				resolve(sprites);
+			};
+		});
+	};
+
+	const drawMesh = (ctx: CanvasRenderingContext2D) => {
+		for (let i = 0; i < width; i++) {
+			for (let j = 0; j < height; j++) {
+				ctx.rect(i * rectSize, j * rectSize, rectSize, rectSize);
+				ctx.stroke();
+			}
+		}
+	};
+
+	const drawSprites = (ctx: CanvasRenderingContext2D) => {
+		if (!sprites) {
+			return;
+		}
+
+		for (let y = 0; y < height; y++) {
+			for (let x = 0; x < width; x++) {
+				const coordinates = {x, y};
+				const status = ground.getCellStatus(coordinates);
+				let module, size, spriteIndex, sprite;
+				switch (status) {
+					case CellStatus.OCCUPIED:
+						if (owner === 'player') {
+							module = battle.findSpaceModule(coordinates);
+							if (!module) {
+								throw new Error('Ошибка отрисовки графики');
+							}
+							size = module.size;
+							spriteIndex = module.getCellIndex(coordinates);
+							sprite = module.vertical
+								? sprites.modules[size][spriteIndex].vertical
+								: sprites.modules[size][spriteIndex].horizontal;
+						} else {
+							sprite = sprites.modules[3][2].vertical;
+						}
+						ctx.drawImage(
+							sprite,
+							x * rectSize,
+							y * rectSize,
+							rectSize,
+							rectSize);
+						break;
+					case CellStatus.BURNING:
+						ctx.drawImage(
+							sprites.burn,
+							x * rectSize,
+							y * rectSize,
+							rectSize,
+							rectSize);
+						break;
```

Comment:
Можно сократить до одного case. Остальные ниже убрать.
```suggestion
					case CellStatus.BURNING:
					case CellStatus.DESTROYED:
					case CellStatus.MISSED:
						const sprite = sprites.`XXX`;
						ctx.drawImage(
							sprite,
							x * rectSize,
							y * rectSize,
							rectSize,
							rectSize);
						break;
```

Reply:
Тут можно в switch выбирать какой sprite рисовать и после уже делать ctx.drawImage.
Можно попробовать вообще без switch, а сделать объект, где CellStatus ключ, а конкретный спрайт значение. Но! В случае CellStatus.OCCUPIED логика сложнее, нужно будет это обработать всё равно.